### PR TITLE
Add "Create" and chat icon to top menu

### LINKED ISSUE #2528: Add "Create" and chat icon to top menu
Please add a "Create" item and a chat icon to the top menu bar to the right hand side of the user name/email.

On clicking the Chat Icon
* open the chat window
* if the chat box is already open, close it when the user clicks the chat icon

Scrolling over "Create" should result in a submenu opening showing:
* Create Skill -> https://susi.ai/skillWizard
* Create Bot -> https://susi.ai/botbuilder/botwizard (or the updated link)

Also compare to Facebook
Screenshot from 2019-07-11 10-09-47

### PR DESCRIPTION
**Pull request by samagragupta**
Fixes #2528

Changes: 
Added a "Create" item and a chat icon to the top menu bar to the right-hand side of the user name/email.

- On clicking the Chat Icon
open the chat window
if the chat box is already open, close it when the user clicks the chat icon
- Scrolling over "Create" should result in a submenu opening show:
Create Skill -> https://susi.ai/skillWizard
Create Bot -> https://susi.ai/botbuilder/botwizard (or the updated link)

Demo Link: https://pr-2602-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 
Capture

### CODE REVIEW

**Comment by mariobehling** [comment]
Clicking on the top chat icon should open the chat box (same as Facebook).

Screenshot from 2019-07-16 19-56-04

**Comment by samagragupta** [comment]
@fragm3 Working fine in my case.
Capture

**Comment by mariobehling** [comment]
Please also switch the dashboard icon with "Create".

**Comment by mariobehling** [comment]
@fragm3 

> Rightmost menu is getting cut:
> Screenshot 2019-07-16 at 11 26 09 PM

Actually the icon disappears completely if there is a very long user name. We could shorten the search box or implement it in a way that it expands when the user adds search terms. We can follow up on this separately.

File: src/components/NavigationBar/index.js [hunk 1]
```diff
@@ -522,6 +560,12 @@ class NavigationBar extends Component {
                     </ListItemText>
                   </MenuItem>
                 )}
+                <IconButton
+                  color="inherit"
+                  onClick={() => history.push('/chat')}
```
Comment (akshatnitd) [inline src/components/NavigationBar/index.js]: Why are you opening chat page here? Check the issue.

**Comment by mariobehling** [comment]
Dashboard icon disappeard: It should be next to "Create" on the right hand side.

**Comment by mariobehling** [comment]
Order from left to right: User - Create - Dashboard icon - Chat icon - Question/Support Icon

**Comment by samagragupta** [comment]
@mariobehling @akshatnitd Done the changes

**Comment by mariobehling** [comment]
> On clicking the Chat Icon
* open the chat window
* if the chat box is already open, close it when the user clicks the chat icon

**Comment by samagragupta** [comment]
@mariobehling Done

**Comment by akshatnitd** [comment]
@mariobehling Do not merge it. There are some issues.

File: src/components/NavigationBar/index.js [hunk 2]
```diff
@@ -165,9 +175,21 @@ class NavigationBar extends Component {
       searchType: this.props.searchType,
       searchSelectWidth: this.getSelectMenuWidth(this.props.searchType),
       showSearchBar: window.innerWidth > 900,
+      chat: false,
     };
   }
 
+  toggleChat = () => {
```
Comment (akshatnitd) [inline src/components/NavigationBar/index.js]: These 2 functions should be used on chat icon click based on screen size- 

Check `MessageSection.react.js` file.
```
  openFullScreen = () => {
    const { actions } = this.props;
    actions.handleChatBubble({
      chatBubble: 'minimised',
    });
    actions.handleTestSkillExample({ testSkillExampleKey: -1 });
    actions.openModal({
      modalType: 'chatBubble',
      fullScreenChat: true,
    });
  };

  toggleChat = () => {
    const { actions, chatBubble } = this.props;
    actions.handleChatBubble({
      chatBubble: chatBubble === 'bubble' ? 'full' : 'bubble',
    });
    actions.handleTestSkillExample({ testSkillExampleKey: -1 });
  };
```

Comment (samagragupta) [reply src/components/NavigationBar/index.js]: I think we can remove this icon when the screen size is small. As there is less space left in the navigation bar.

Comment (samagragupta) [reply src/components/NavigationBar/index.js]: @mariobehling What are your views?

Comment (akshatnitd) [reply src/components/NavigationBar/index.js]: No, the space management and UI for navbar will be done by me in a new PR. Please make the requested changes.

Comment (samagragupta) [reply src/components/NavigationBar/index.js]: @akshatnitd Done

File: src/components/NavigationBar/index.js [hunk 3]
```diff
@@ -165,9 +175,40 @@ class NavigationBar extends Component {
       searchType: this.props.searchType,
       searchSelectWidth: this.getSelectMenuWidth(this.props.searchType),
       showSearchBar: window.innerWidth > 900,
+      chat: false,
+      width: window.innerWidth,
```
Comment (akshatnitd) [inline src/components/NavigationBar/index.js]: Why do you want it as state?

File: src/components/NavigationBar/index.js [hunk 4]
```diff
@@ -522,6 +593,12 @@ class NavigationBar extends Component {
                     </ListItemText>
                   </MenuItem>
                 )}
+                <IconButton
+                  color="inherit"
+                  onClick={width < 500 ? this.openFullScreen : this.toggleChat}
```
Comment (akshatnitd) [inline src/components/NavigationBar/index.js]: Directly use `window.innerWidth` here.

Comment (samagragupta) [reply src/components/NavigationBar/index.js]: @akshatnitd Done

**Comment by samagragupta** [comment]
@akshatnitd Done

**Comment by samagragupta** [comment]
@akshatnitd Sorry, forget to remove updateWindowDimensions.
Removed that

File: src/components/NavigationBar/index.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ import CircleImage from '../shared/CircleImage';
 import { bindActionCreators } from 'redux';
 import uiActions from '../../redux/actions/ui';
 import skillActions from '../../redux/actions/skills';
```
Comment (fragm3) [inline src/components/NavigationBar/index.js]: Make it: skillsAction

File: src/components/NavigationBar/index.js [hunk 6]
```diff
@@ -570,12 +644,16 @@ function mapStateToProps(store) {
...
     ...store.skills,
+    testSkillExampleKey: store.skill.testSkillExampleKey,
   };
 }
 
 const mapDispatchToProps = dispatch => {
   return {
-    actions: bindActionCreators({ ...skillActions, ...uiActions }, dispatch),
+    actions: bindActionCreators(
+      { ...skillActions, ...uiActions, ...skillAction },
```
Comment (fragm3) [inline src/components/NavigationBar/index.js]: skillActions --> skillsActions

**Comment by samagragupta** [comment]
@fragm3 Fixed.

**Comment by samagragupta** [comment]
@akshatnitd Fixed that

**Pull request merged**